azure-maven-plugin/src/main/java/com/microsoft/azure/plugin/login/LoginMojo.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.plugin.login;
+
+import com.microsoft.azure.AzureEnvironment;
+import com.microsoft.azure.auth.AzureAuthHelper;
+import com.microsoft.azure.auth.AzureCredential;
+
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+
+/**
+ * Goal to login to azure.
+ */
+@Mojo(name = "login", inheritByDefault = true, aggregator = true)
+public class LoginMojo extends AbstractMojo {
+
+    @Parameter(property = "devicelogin")
+    public boolean devicelogin;
+
+    @Parameter(property = "environment")
+    public String environment;
+
+    @Override
+    public void execute() throws MojoFailureException {
+        final AzureEnvironment env = AzureAuthHelper.getAzureEnvironment(environment);
+        getLog().info("Begin to login");
```

Comment:
If the log text is inferring a procedure is going on, append `...` at the end.

Reply:
done